### PR TITLE
get neighbor() from kad routing table instead of local peerstore

### PR DESCRIPTION
**Pull request by Liuhaai**
linked to https://github.com/iotexproject/iotex-core/issues/2729
linked to https://github.com/iotexproject/iotex-core/issues/2631

### CODE REVIEW

**Review by raullenchai — APPROVED** [review]

**Comment by raullenchai** [comment]
Why the test failed?

**Comment by Liuhaai** [comment]
> Why the test failed?

There is no guarantee that peers from routing table take one third of all peers in the network

File: p2p.go [hunk 1]
```diff
@@ -100,6 +98,7 @@ var (
 		EnableRateLimit:          false,
 		RateLimit:                DefaultRatelimitConfig,
 		PrivateNetworkPSK:        "",
+		DHTProtocolName:          "/iotex",
```
Comment (dustinxie) [inline p2p.go:101]: 1. call it `ProtocolName`? now it used DHT, in theory can be other protocol
2. move to line 48

File: p2p.go [hunk 2]
```diff
@@ -213,6 +212,16 @@ func PrivateNetworkPSK(privateNetworkPSK string) Option {
 	}
 }
 
+// DHTProtocolName returns the prefix of dht protocol. MainNet uses "/iotex", while testNet uses "/iotexTestNet"
+func DHTProtocolName(chainID uint32) Option {
+	return func(cfg *Config) error {
+		if chainID != 1 {
+			cfg.DHTProtocolName = "/iotexTestNet"
```
Comment (dustinxie) [inline p2p.go:219]: let's use "iotex2" for chainID = 2, "iotex7" for chainID = 7 for generic case
then in the future, we don't need to modify here if someone starts up a new sub-network

File: p2p.go [hunk 3]
```diff
@@ -21,16 +21,16 @@ import (
...
 	"github.com/libp2p/go-libp2p-core/pnet"
 	"github.com/libp2p/go-libp2p-core/protocol"
-	"github.com/libp2p/go-libp2p-discovery"
-	"github.com/libp2p/go-libp2p-kad-dht"
-	"github.com/libp2p/go-libp2p-pubsub"
-	"github.com/libp2p/go-libp2p-secio"
+	discovery "github.com/libp2p/go-libp2p-discovery"
+	dht "github.com/libp2p/go-libp2p-kad-dht"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	secio "github.com/libp2p/go-libp2p-secio"
```
[the remark this answers is not in the document]
Comment (dustinxie) [reply p2p.go:33]: if this is the result of running `go fmt`, we should accept
otherwise, it is unnecessary change, right?

File: p2p.go [hunk 4]
```diff
@@ -48,6 +46,7 @@ type (
 
 	// Config enumerates the configs required by a host
 	Config struct {
+		ProtocolName             protocol.ID     `yaml:"dhtProtocol"`
```
Comment (CoderZhi) [inline p2p.go:49]: we'd do our best to have keep the consistency of the parameter name and yaml key. so, yaml:"protocolName"

File: p2p_test.go [hunk 5]
```diff
@@ -88,7 +88,7 @@ func TestUnicast(t *testing.T) {
 
 	for i, host := range hosts {
 		neighbors := host.Neighbors(ctx)
-		require.True(t, len(neighbors) >= n/3)
+		require.True(t, len(neighbors) > 0)
```
Comment (dustinxie) [inline p2p_test.go:91]: let's keep `n/3`

reason is that `> 0` is naively satisfied (each node will connect to bootstrap node), last time upgrade p2p has a bug (DHT is not set to server mode), using `> 0` failed to detect such bug, so it is changed to `n/3`

if test fails due to this, just re-submit the test

File: p2p.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (Liuhaai) [reply p2p.go:34]: results of `go fmt`

**Comment by bighugo33** [comment]
- [ ]

**Pull request merged**